### PR TITLE
refactor(helm): add control for securityContext

### PR DESCRIPTION
**Pull request by zvlb**
Signed-off-by: Zemtsov Vladimir <zemtsov.v@mail366.com>

Small feature for control securityContext in helm

Local test:
```bash
~ kind create cluster
...


~ k cluster-info
Kubernetes control plane is running at https://127.0.0.1:60977
CoreDNS is running at https://127.0.0.1:60977/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.


~ helm install capsule . -n capsule-system --create-namespace
NAME: capsule
LAST DEPLOYED: Fri Feb 10 09:57:02 2023
NAMESPACE: capsule-system
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
  ...

~ k get deployments.apps -n capsule-system capsule-controller-manager -o yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  ...
  name: capsule-controller-manager
  namespace: capsule-system
  ...
spec:
  ...
  template:
    ...
    spec:
      containers:
      - ...
        ...
        securityContext:
          allowPrivilegeEscalation: false
          capabilities:
            drop:
            - ALL
          readOnlyRootFilesystem: true
        ...
      ...
      securityContext:
        runAsGroup: 1002
        runAsNonRoot: true
        runAsUser: 1002
      ...



~ k create -f - << EOF
apiVersion: capsule.clastix.io/v1beta2
kind: Tenant
metadata:
  name: oil
spec:
  owners:
  - name: alice
    kind: User
EOF
tenant.capsule.clastix.io/oil created


~ k get tenants
NAME   STATE    NAMESPACE QUOTA   NAMESPACE COUNT   NODE SELECTOR   AGE
oil    Active                     0                                 4s


~ ../../hack/create-user.sh alice oil
creating certs in TMPDIR /var/folders/c7/78_t4pvj5zd2qrcwgq05h7_40000gn/T/tmp.Sr8bOiQ3
merging groups /O=capsule.clastix.io
Generating RSA private key, 2048 bit long modulus
......................................+++
........+++
e is 65537 (0x10001)
certificatesigningrequest.certificates.k8s.io/alice-oil created
certificatesigningrequest.certificates.k8s.io/alice-oil approved
kubeconfig file is: alice-oil.kubeconfig
to use it as alice export KUBECONFIG=alice-oil.kubeconfig


~ export KUBECONFIG=alice-oil.kubeconfig
~ k create namespace oil-production
namespace/oil-production created

~ k create namespace oil-development
namespace/oil-development created


~ export KUBECONFIG=
~ k get tenant oil
NAME   STATE    NAMESPACE QUOTA   NAMESPACE COUNT   NODE SELECTOR   AGE
oil    Active                     2                                 38s
```

### CODE REVIEW

**Comment by zvlb** [comment]
This PR continuation of this - https://github.com/clastix/capsule/pull/693

SecurityContext I take from this issue - https://github.com/clastix/capsule/issues/680

**Review by prometherion — APPROVED** [review]

**Comment by prometherion** [comment]
Astonishing work, @zvlb! <3

@oliverbaehler may I ask for your review on this?

File: charts/capsule/values.yaml [hunk 1]
```diff
@@ -77,6 +77,21 @@ podAnnotations: {}
 # -- Set the priority class name of the Capsule pod
 priorityClassName: '' # system-cluster-critical
 
+# -- Set the securityContext for the Capsule pod
+podSecurityContext:
```
Comment (oliverbaehler) [inline charts/capsule/values.yaml:81]: Add `seccompProfile` by default


```suggestion
podSecurityContext:
  seccompProfile:
    type: "RuntimeDefault"

```

**Review by oliverbaehler — CHANGES_REQUESTED** [review]
Thanks for your contribution. Please look at my suggested change and retest everything :)

**Comment by zvlb** [comment]
I added default seccompProfile and retest capsule. All work good

**Review by oliverbaehler — APPROVED** [review]
/lgtm

**Comment by prometherion** [comment]
> What's the behavior for you? (Just want to make sure this doesn't break the default deployment) :)

It's something I experienced too, it happens time to time while generating the webhook certificates.

We should increase the memory of the Controller to avoid this.

**Comment by zvlb** [comment]
Hi. I have a question.
Why step "release" was skipped?

**Comment by prometherion** [comment]
Because I merged the PR and hadn't pushed the Helm tag:

https://github.com/clastix/capsule/blob/770ad22170d5484de7093bdcf8aa27178e9e95fe/.github/workflows/helm.yml#L70-L86

Please, note line nr. 71

Just pushed the related tag and the new Chart version is available, remember to issue `helm repo update` to get your local cache updated!

**Comment by zvlb** [comment]
TY!

**Pull request merged**